### PR TITLE
Pilot agent will not watch cert file when auth is disabled

### PR DESCRIPTION
**Pull request by myidpt**
This fixes https://github.com/istio/issues/issues/169

### CODE REVIEW

**Comment by wattli** [comment]
Should we change the title to pilot agent?

File: pilot/proxy/envoy/watcher.go [hunk 1]
```diff
@@ -55,23 +55,25 @@ type CertSource struct {
...
+	certs      []CertSource
+	pilotSAN   []string
+	enableAuth bool
 }
 
 // NewWatcher creates a new watcher instance from a proxy agent and a set of monitored certificate paths
 // (directories with files in them)
 func NewWatcher(config meshconfig.ProxyConfig, agent proxy.Agent, role model.Node,
-	certs []CertSource, pilotSAN []string) Watcher {
+	certs []CertSource, pilotSAN []string, enableAuth bool) Watcher {
```
Comment (kyessenov) [inline pilot/proxy/envoy/watcher.go]: I think breaks ingress cert renewal?

Comment (myidpt) [reply pilot/proxy/envoy/watcher.go]: That's a good point. I think we may need to differentiate the file path as well ...

Comment (myidpt) [reply pilot/proxy/envoy/watcher.go]: I will move the logic out the the cmd main function.

**Comment by myidpt** [comment]
@andraxylia @kyessenov Can you give /lgtm if this PR looks good to you? Thanks.

**Comment by andraxylia** [comment]
Not sure why the bot did not kick in within an hour.

**Comment by ldemailly** [comment]
this should make it to 0.5.0 it's a very big improvement for some customers

**Pull request merged**